docs/DeveloperGuide.md

```diff
@@ -1074,22 +1141,21 @@ Use case ends.
 
 { end of `requirements#use_cases` written by: Zhang Sheng Yang }
 
-{ start of `requirements#non_functional_requirements` written by: ___________ }
+{ start of `requirements#non_functional_requirements` written by: Ngoh Wei Yue }
 
 ### 10.4. Non-functional requirements
 
-1. Should work on any _mainstream OS_ as long as it has Java `11` or above installed.
-1. Should be able to hold up to 1000 students without a noticeable sluggishness in performance for typical usage.
-1. A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
-1. Should be for a single user and not a multi-user product.
-1. Should not depend on remote servers.
-1. Should be able to function without having to rely on being connected to a network.
-1. Should be able to work without requiring an installer.
-1. Should not contain very large file sizes (JAR files - 100Mb, PDF files - 15Mb/file).
+1. This product should work on any _mainstream OS_ as long as it has Java `11` or above installed.
+1. This product should be able to hold up to 500 students and 500 sessions without a noticeable sluggishness in performance for typical usage.
+1. This product should be optimised for users who can type regular English text (i.e. not code, not system admin commands) fast and prefer typing over other means of input (e.g., mouse).
+1. This product should be for a single user and should not be a multi-user product.
+1. This product should not depend on remote servers.
+1. This product should be able to function without being connected to a network.
+1. This product should be able to work without requiring an installer.
+1. Data generated by the product should be stored locally in human-editable file.
+1. This product Should not contain very large file sizes (JAR files - 100Mb, PDF files - 15Mb/file).
```

Comment:
Maybe "Should" should be de-capitalized here?

Reply:
Good spot! Thank you.